### ISSUE TITLE
[WCAG 2.2 Draft Feedback] Success Criterion 2.5.7 Dragging

### ISSUE THREAD

**Issue by dshoukry** [w3c/wcag#1454]
“Success Criterion 2.5.7 Dragging (Level AA): All functionality that uses a dragging movement for operation can be operated by a single pointer without dragging, unless dragging is essential.”

Adding more examples and clarification to this SC may help to solidify successful implementation of this requirement. We were confused about how this is different from WCAG 2.1 SC 2.5.1 Pointer Gestures and interpreted that most scenarios would likely meet the essential exception. Other comments include: requests to clarifying common essential approved exceptions for dragging besides drawing apps and also providing a few examples of interactions that are incorrectly assumed to be essential approved exceptions, a suggestion to clarify that the exemption covers interactions essential to the task, not essential to a particular interaction design of the task, and a question about a sufficient technique.

Please find detailed specifics covered in our [2.5.7 Dragging (Level AA) Google Doc](https://docs.google.com/document/d/1mOgGI7bdgTqsEYhuOlA9yiWGMU06ONrgL0KbGflYKJw/edit).

**Comment by dshoukry** [comment]
As requested, I just replaced the Google Doc link with a version with expanded access (didn't make any content changes). Sorry for the access issues with the previous link.

**Comment by alastc** [comment]
Hi @dshoukry,

I have a couple of comments and responses to your comments:

> Other than drawing apps, what are some common essential approved exceptions for dragging? ...
> It might also be helpful to explicitly clarify that the exemption covers interactions essential to the task, not essential to a particular interaction design of the task.

We do have a linked definition of essential, which is "if removed, would fundamentally change the information or functionality of the content, and information and functionality cannot be achieved in another way that would conform".
That definition is used quite a lot, I'd rather not add to the SC text for that. Examples in the Understanding doc are fine though, and could include a line about it being essential to the task rather than the interaction.

In a dragging context, needing to enter a handwriting sample is something that requires the information in a particular format and could not be achieved in another way. (You could argue that physical signatures are not needed given that we now have digital signatures now, so I avoid that as an example.)

> We were confused about how this is different from WCAG 2.1 SC 2.5.1 Pointer Gestures and interpreted that most scenarios would likely meet the essential exception.

We have another issue where we are dealing with the overlap. Could you provide any scenarios that you thought would fit the essential exception? 

Regarding Dragon's features to do drag and drop, that's good to know (I'm sure other people in the group have experience with that). I think the idea is that this would help people who are using a 'pointer' (probably a finger) but do not (necessarily) use voice access. E.g. we've had a participant in research with a significant mobility impairment and they used the knuckle of their wrist to tap, but dragging was next to impossible.

**Comment by alastc** [comment]
For the essential examples aspect, I've expanded the text in #1612 to cover the comments here.

**Comment by alastc** [comment]
Draft response:
<hr>

For the first comment:
> Other than drawing apps, what are some common essential approved exceptions for dragging?
>
> Can the group also please provide a few examples of interactions that are incorrectly assumed to be essential approved exceptions? It might also be helpful to explicitly clarify that the exemption covers interactions essential to the task, not essential to a particular interaction design of the task.

PR #1612 includes examples in both directions.

Comment 2:
> For AT to assist, there likely needs to be something in the UI that the AT can pick up (e.g., handles that can be activated via switch that trigger movement)
> 
> Dragon NaturallySpeaking has mouse drag commands as discussed in the following PDF: https://www.nuance.com/content/dam/nuance/en_us/collateral/dragon/command-cheat-sheet/ct-dragon-naturally-speaking-en-us.pdf
> 
> Coolfields has a blog post with a video, and transcript demonstrating this at: https://www.coolfields.co.uk/2014/05/dragon-naturally-speaking-video-10-drag-drop/
> 
> Seems like it would also work with modes that show up in mouseover.
Show less

The criterion is intended to benefit people who are using a 'pointer' (probably a finger) but do not use voice access. For that scenario there would need to be some UI available on tap. E.g. an element could be dragged, or it could be tapped and up/down/left/right buttons appear.

Comment 3, on the technique linked:
> If an object is locked to the pointer, placed in the location, and then deselected, would that sufficiently negate the need to hold and drag? This could also be done with keyboard navigation.

If it requires constant contact with the surface, then no, that would be a dragging movement.

**Comment by alastc** [comment]
The above response was accepted by the group today.
